build-support/bin/check_pants_pex_abi.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+# Check that the ./pants.pex was built using the passed abi specification.
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+source ${REPO_ROOT}/build-support/common.sh
+CHECK_FOLDER="${REPO_ROOT}/pants_pex_abi_check"
+
+EXPECTED_ABI="$1"
+if [ -z "${EXPECTED_ABI}" ]; then
+  die "Must pass the expected abi as an argument. E.g. 'abi3' or 'cp27mu'."
+fi
+
+if [ ! -f "${REPO_ROOT}/pants.pex" ]; then
+  die "pants.pex not found in the repository root! Run './build-support/bin/ci.sh -b'."
+fi
+
+function parse_wheel_abi() {
+  # See https://www.python.org/dev/peps/pep-0425/#use for how wheel names are defined.
+  wheel_filename="$1"
+  IFS='-' read -r -a components <<< "${wheel_filename}"
+  # Check if optional build tag
+  if [ ${#components[@]} -eq 6 ]; then
+    abi="${components[4]}"
+  else
+    abi="${components[3]}"
+  fi
+  echo "${abi}"
+}
+
+function cleanup() {
+  rm -rf "${CHECK_FOLDER}"
+}
+
+# Extract pex
```

Comment:
You don't need to extract the entire pex, since you only care about one file in it - PEX-INFO. And you don't even need to extract that to disk, this will send just that file to stdout:

```unzip -p pants.pex PEX-INFO```

So you can get rid of the CHECK_FOLDER and its cleanup, and just pipe this directly to the `jq` command (see below).